runtime/lua/vim/lsp/semantic_tokens.lua

```diff
@@ -36,9 +36,7 @@ local util = require('vim.lsp.util')
 local STHighlighter = { active = {} }
 
 ---@private
-local function binary_search(tokens, line)
-  local lo = 1
-  local hi = #tokens
+local function lower_bound(tokens, line, lo, hi)
```

Comment:
Should the docstring at least mention "binary search"? That's a helpful hint.

same for upper_bound

Reply:
Yes, it should. Fixed.